aptos-move/framework/aptos-framework/sources/datastructures/ordered_map.move

```diff
@@ -542,18 +542,35 @@ module aptos_std::ordered_map {
         dk: |K|,
         dv: |V|
     ) {
-        let (keys, values) = to_vec_pair(self);
-        vector::destroy(keys, |_k| dk(_k));
-        vector::destroy(values, |_v| dv(_v));
+        let (keys, values) = self.to_vec_pair();
+        keys.destroy(|_k| dk(_k));
+        values.destroy(|_v| dv(_v));
     }
 
     /// Apply the function to a reference of each key-value pair in the table.
-    public inline fun for_each_ref<K, V>(self: &OrderedMap<K, V>, f: |&K, &V|) {
-        let iter = self.new_begin_iter();
-        while (!iter.iter_is_end(self)) {
-            f(iter.iter_borrow_key(self), iter.iter_borrow(self));
-            iter = iter.iter_next(self);
-        }
+    public inline fun for_each_ref<K: copy + drop, V>(self: &OrderedMap<K, V>, f: |&K, &V|) {
+        // This implementation is innefficient: O(n) for next_key / borrow lookups every time,
```

Comment:
Add the comments on the time complexity to the function comment so that the generated doc will have the information in the header.

Reply:
added